src/java/org/apache/cassandra/config/DatabaseDescriptor.java

```diff
@@ -4763,4 +4756,159 @@ public static OptionalDouble getSeverityDuringDecommission()
+    {
+        Property property = ofNullable(confValueAccessors.get(name))
+                            .map(p -> {
+                                // The case when replacement property is used and match the property's name.
+                                if (p instanceof Replacement.ReplacementProperty)
+                                    return ((Replacement.ReplacementProperty) p).delegate();
+                                else
+                                    return p;
+                            })
+                            .orElseThrow(() -> new PropertyNotFoundException(name));
+        if (!(property instanceof ListenableProperty))
+            throw new ConfigurationException(String.format("Property '%s' is not listenable.", name));
+
+        if (clazz.equals(property.getType()))
+            return adder.apply((ListenableProperty<Config, T>) property);
+        else
+            throw new ConfigurationException(String.format("Listener type '%s' doesn't match the property '%s' type '%s'.",
+                                                           clazz.getSimpleName(), name, clazz.getSimpleName()));
+    }
+
+    public static Object getProperty(String name)
+    {
+        return ofNullable(confValueAccessors.get(name)).orElseThrow(() -> new PropertyNotFoundException(name))
+                                                       .get(conf);
+    }
+
+    public static String getStringProperty(String name)
+    {
+        Property property = ofNullable(confValueAccessors.get(name))
+                            .orElseThrow(() -> new PropertyNotFoundException(name));
+        return property.getType().equals(String.class) ?
+               (String) property.get(conf) :
+               propertyToStringConverter().convertNullable(property.get(conf));
+    }
+
+    /**
+     * Set configuration property for the given name to {@link #conf} if a safe manner
+     * with handling internal Cassandra exceptions.
+     *
+     * @param name Property name.
+     * @param value Property value.
+     */
+    public static void setProperty(String name, Object value)
+    {
+        try
+        {
+            Property property = ofNullable(confValueAccessors.get(name)).orElseThrow(() -> new PropertyNotFoundException(name));
+            if (property.getAnnotation(Mutable.class) == null)
+                throw new ConfigurationException("Property is read-only: " + property.getName());
+
+            if (value == null)
+            {
+                property.set(conf, null);
+                return;
+            }
+
+            if (property.getType().isAssignableFrom(value.getClass()))
+                property.set(conf, value);
+            else if (String.class.equals(value.getClass()))
+                property.set(conf, propertyFromStringConverter(primitiveToWrapper(property.getType())).convert((String) value));
```

Comment:
push this into `SettingsTable`, you are leaking details cross classes, we should keep this very simple